### ISSUE TITLE
Segmentation fault while updating 

### ISSUE THREAD

**Issue by eliuha** [mstinaff/PMS_Updater#69]
Running the script from BSD Jail on True NAS

```

# ./PMS_Updater.sh -v
Using URL https://plex.tv/api/downloads/5.json
Searching https://plex.tv/api/downloads/5.json for the FreeBSD download URL .....Segmentation fault
Could not find a FreeBSD download link on page https://plex.tv/api/downloads/5.json?channel=plexpass&X-Plex-Token=<token masked>
```

**Comment by Logan-Fouts** [comment]
I too am receving this error when running that script. So there must be some real issue of some sort

**Comment by eliuha** [comment]
A workaround for me was manually downloading the tar and installing it using the -l option, but that's not the best way we could do

**Comment by MAGICCC** [comment]
I don't really understand why https://github.com/mstinaff/PMS_Updater/blob/master/PMS_Updater.sh#L152 would fail sometimes...

**Comment by adostrom** [comment]
I have been getting this error, too.

**Comment by lordviceroy** [comment]
I realize this probably won't be super helpful, but I had this issue a while back, and I vaguely recall it being somehow related to the URL/domain that the script was using to fetch the update. In the interim, eliuha's suggestion of manually downloading the update and using -l should work for most people.

That aside, I see a bunch of stale PR's that got auto closed, so this script might not be maintained at the moment. If someone figures out a fix, consider forking it.

**Comment by KirkKirk** [comment]
> A workaround for me was manually downloading the tar and installing it using the -l option, but that's not the best way we could do

@eliuha Could you kindly provide a detailed explanation of this process, broken down step by step? 
Thanks

**Comment by davidsch1992** [comment]
@KirkKirk 
The script is working for me now on my TrueNAS machine. 

Here is a guide to do it manually:

[Manual update with PMS script](https://github.com/mstinaff/PMS_Updater/issues/67#issuecomment-1672282328)

**Comment by MAGICCC** [comment]
> That aside, I see a bunch of stale PR's that got auto closed, so this script might not be maintained at the moment. If someone figures out a fix, consider forking it.

Yes because I added a GitHub action for it, issues which are years old without updates can be closed. At the moment my installations are working fine with this script. Do not hasitate to help the community with updating parts of the script if you see some issues

**Comment by Lerwox** [comment]
I solved this issue with the installation of CURL.
Segmentation fault was caused by the Certificate Authority of CloudFlare that was not a part of original FreeBSD CAs

**Comment by MAGICCC** [comment]
@Lerwox 
Could you try to run fetch with `--no-verify-peer` parameters?
`fetch --no-verify-peer -o /tmp URL`

Maybe we can add a parameter to ignore SSL certs (yes it's not nice)

**Comment by Lerwox** [comment]
@MAGICCC 
I solved the problem by installing curl (`pkg install curl`), which added CloudFlare's Certificate Authority. 
Unfortunately, I don't have a backup to test with `--no-verify-peer`, sorry.

I'd like to point out that I installed CURL but I don't use it, I didn't change anything in the script. However, installing CURL added the missing CA to the jail files. So, whether it's `wget`, `curl` or `fetch`, it works now
